File: device-sdk/src/main/java/io/oddworks/device/metric/OddAppInitMetric.java
package io.oddworks.device.metric;

public class OddAppInitMetric extends OddMetric {
    private static final String TAG = OddAppInitMetric.class.getSimpleName();

    private static String appInitAction = ACTION_APP_INIT;

    public OddAppInitMetric() {

    }

    public static void setAction(String action) {
        appInitAction = action;
    }

    public static void setEnabled(String mEnabled) {
        // Don't allow AppInit to be disabled
    }

    @Override
    public String getAction() {
        return appInitAction;
    }

    @Override
    public boolean getEnabled() {
        return true;
    }
}

File: device-sdk/src/main/java/io/oddworks/device/metric/OddAppInitMetric.kt
package io.oddworks.device.metric

import org.json.JSONObject

class OddAppInitMetric(contentType: String? = null, contentId: String? = null, title: String? = null, meta: JSONObject? = null) : OddMetric(contentType, contentId, title, meta) {

    override val action: String
        get() = OddAppInitMetric.action


    override val enabled: Boolean
        get() = OddAppInitMetric.enabled

    companion object {
        private val TAG = OddAppInitMetric::class.java.simpleName
        var action = OddMetric.ACTION_APP_INIT
        var enabled = true
    }
}